### PR TITLE
Number format read fix

### PR DESCRIPTION
**Pull request by artiz**
# PR Details

We are using excelize in our ETL project to parse variety XLSX files from our customers and had found that some XLSX files generated by ERP systems contain incorrect number format definitions that leads to panic in excelize library.

We have added safety check to avoid panic on XSLX file read

## How Has This Been Tested

Added new UT with example file

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: crypt_test.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 func TestEncrypt(t *testing.T) {
 	f, err := OpenFile(filepath.Join("test", "encryptSHA1.xlsx"), Options{Password: "password"})
 	assert.NoError(t, err)
-	assert.EqualError(t, f.SaveAs(filepath.Join("test", "BadEncrypt.xlsx"), Options{Password: "password"}), "not support encryption currently")
+	assert.EqualError(t, f.SaveAs(filepath.Join("test", "TestBadEncrypt.xlsx"), Options{Password: "password"}), "not support encryption currently")
```
Comment (xuri) [inline crypt_test.go:22]: I suggest to keep the spreadsheet name of them. In fact, I have a program offline that will batch verify the generated file with `Test` prefix can be opened correctly by Office applications :)

**Review by xuri — CHANGES_REQUESTED** [review]
Hi @artiz, thanks for your PR. I've left some comments.

File: cell.go [hunk 2]
```diff
@@ -762,9 +762,14 @@ func (f *File) formattedValue(s int, v string) string {
 		return v
 	}
 	styleSheet := f.stylesReader()
+
```
[the remark this answers is not in the document]
Comment (artiz) [reply cell.go:765]: got it, will refactor

File: cell_test.go [hunk 3]
```diff
@@ -321,11 +321,39 @@ func TestFormattedValue(t *testing.T) {
...
 	numFmtID := 5
 	f.Styles.CellXfs.Xf = append(f.Styles.CellXfs.Xf, xlsxXf{
 		NumFmtID: &numFmtID,
 	})
-	v = f.formattedValue(1, "43528")
+	v = f.formattedValue(2, "43528")
+	assert.Equal(t, "43528", v)
+
+	// formatted value with invalid number format ID
+	assert.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (artiz) [reply cell_test.go:332]: yep, copy-paste issue

File: cell_test.go [hunk 4]
```diff
@@ -321,11 +321,39 @@ func TestFormattedValue(t *testing.T) {
 
 	// formatted value with no built-in number format ID
 	assert.NoError(t, err)
-	f.Styles.NumFmts = nil
```
[the remark this answers is not in the document]
Comment (artiz) [reply cell_test.go:324]: will do

**Comment by xuri** [comment]
LGTM, thanks for your PR.

**Pull request merged**